From 433796d860326e4ffa344fd2b361f332470ad357 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@digia.com>
Date: Fri, 21 Feb 2014 15:12:45 +0100
Subject: [PATCH] Fix Qt WebKit build on architectures without JIT support

The fallback LLINT interpreter was triggering YARR JIT which would
fail to compile and also trigger ENABLE_ASSEMBLER which would trigger
even more not to compile. YARR JIT is only supported on platforms with
valid JIT or native LLINT support. So we must avoid it when the fallback
is used

Task-number: QTBUG-36969
Change-Id: Ie94bcb316e192feb33ab528ddc3a328af154cebf
---
 Source/JavaScriptCore/bytecode/CodeBlock.cpp | 4 ++++
 Source/WTF/wtf/Platform.h                    | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/Source/JavaScriptCore/bytecode/CodeBlock.cpp b/Source/JavaScriptCore/bytecode/CodeBlock.cpp
index 904e40a..efe4424 100644
--- a/Source/JavaScriptCore/bytecode/CodeBlock.cpp
+++ b/Source/JavaScriptCore/bytecode/CodeBlock.cpp
@@ -2947,8 +2947,12 @@ void CodeBlock::countReoptimization()
 
 unsigned CodeBlock::numberOfDFGCompiles()
 {
+#if ENABLE(JIT)
     ASSERT(JITCode::isBaselineCode(getJITType()));
     return (JITCode::isOptimizingJIT(replacement()->getJITType()) ? 1 : 0) + m_reoptimizationRetryCounter;
+#else
+    return 0;
+#endif
 }
 
 int32_t CodeBlock::codeTypeThresholdMultiplier() const
diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 1ad7a47..cf80b76 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -889,7 +889,7 @@
 #define ENABLE_REGEXP_TRACING 0
 
 /* Yet Another Regex Runtime - turned on by default for JIT enabled ports. */
-#if !defined(ENABLE_YARR_JIT) && (ENABLE(JIT) || ENABLE(LLINT_C_LOOP)) && !(OS(QNX) && PLATFORM(QT))
+#if !defined(ENABLE_YARR_JIT) && !ENABLE(LLINT_C_LOOP) && !(OS(QNX) && PLATFORM(QT))
 #define ENABLE_YARR_JIT 1
 
 /* Setting this flag compares JIT results with interpreter results. */
-- 
1.9.0

